pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -704,10 +737,39 @@ func (r *ReconcileClusterPool) createCluster(
 			continue
 		}
 		poolRef := poolReference(clp)
+		if clp.Spec.Inventory != nil {
+			poolRef.ClusterDeploymentCustomizationRef = &corev1.LocalObjectReference{Name: cdc.Name}
+		}
 		cd.Spec.ClusterPoolRef = &poolRef
 		lastIndex := len(objs) - 1
 		objs[i], objs[lastIndex] = objs[lastIndex], objs[i]
 	}
+	// Apply inventory customization
+	if clp.Spec.Inventory != nil {
+		for _, obj := range objs {
+			var ok bool
+			if ics, ok = obj.(*corev1.Secret); ok {
+				installConfig, ok := ics.StringData["install-config.yaml"]
+				if ok {
+					newInstallConfig, err := r.getCustomizedInstallConfig([]byte(installConfig), cdc, logger)
+					if err != nil {
+						r.setInventoryValidCondition(clp, false, fmt.Sprintf("failed to customize with %s", cdc.Name), logger)
```

Comment:
what happens when multiple customizations fail for clusterpool? How do we display that info on `InventoryInvalid` condition?

Reply:
Also error handlng is missing

Reply:
> what happens when multiple customizations fail for clusterpool? How do we display that info on InventoryInvalid condition?

It is really hard to make a meaningful message for invalid inventory if won't keep track the individual status of customizations in the clusterpool.

Reply:
https://github.com/openshift/hive/pull/1687